main.tf

```diff
@@ -116,22 +125,46 @@ resource "aws_instance" "default" {
   tags = "${module.label.tags}"
 }
 
+resource "tls_private_key" "ssh" {
+  algorithm = "RSA"
+}
+
+##
+## Create keypair if one isn't provided
+##
+
+locals {
+  key_pair_path = "${var.generate_ssh_key_pair == "true" && signum(length(var.ssh_key_pair)) == 0 && signum(var.instance_count) == 1 ? "${path.cwd}/${element(aws_key_pair.ssh.*.key_name, 0)}.pem" : "" }"
+}
+
+resource "aws_key_pair" "ssh" {
```

Comment:
Any reason why we're not using this module: https://github.com/cloudposse/terraform-aws-key-pair

Reply:
no reason other than i didn't know there was a module.
i can update to use it.